src/driftpy/dlob/DLOB.py

```diff
@@ -0,0 +1,1067 @@
+    ):
+        self.update_resting_limit_orders(slot)
+
+        if order.base_asset_amount == cumulative_base_asset_amount_filled:
+            self.delete(order, user_account, slot)
+            return
+        
+        if order.base_asset_amount_filled == cumulative_base_asset_amount_filled:
+            return
+        
+        new_order = copy.deepcopy(order)
+
+        new_order.base_asset_amount_filled = cumulative_base_asset_amount_filled
+
+        self.get_list_for_order(order, slot).update(new_order, user_account)
+
+
+        if on_update is not None and callable(on_update):
+            on_update()
+    
+    def delete(
+        self,
+        order: Order,
+        user_account: Pubkey,
+        slot: int,
+        on_delete: Optional[OrderBookCallback] = None
+    ):
+        if is_variant(order.status, 'Init'):
+            return
+        
+        self.update_resting_limit_orders(slot)
+
+        self.get_list_for_order(order, slot).remove(order, user_account)
+
+        if on_delete is not None and callable(on_delete):
+            on_delete()
+
+    def clear(self):
+        for market_type in self.open_orders.keys():
+            self.open_orders.get(market_type).clear()
+
+        self.open_orders.clear()
+
+        for market_type in self.order_lists.keys():
+            for market_index in self.order_lists.get(market_type).keys():
+                node_lists: MarketNodeLists = self.order_lists.get(market_type).get(market_index)
+
+                for side in vars(node_lists).keys():
+                    for order_type in getattr(node_lists, side, {}).keys():
+                        getattr(node_lists, side)[order_type].clear()
+        
+        self.order_lists.clear()
+
+        self.max_slot_for_resting_limit_orders = 0
+
+        self.init()
+
+    def trigger(
+        self,
+        order: Order,
+        user_account: Pubkey,
+        slot: int,
+        on_trigger: Optional[OrderBookCallback] = None
+    ):
+        if is_variant(order, 'Init'):
+            return
+        
+        self.update_resting_limit_orders(slot)
+
+        if is_triggered(order):
+            return
+        
+        market_type = market_type_to_string(order.market_type)
+
+        trigger_list = self.order_lists.get(market_type).get(order.market_index) \
+            .trigger['above' if is_variant(order.trigger_condition, 'above') else 'below']
+        trigger_list.remove(order, user_account)
+
+        self.get_list_for_order(order, slot).insert(order, market_type, user_account)
+
+        if on_trigger is not None and callable(on_trigger):
+            on_trigger()
+
+    def handle_order_record(self, record: OrderRecord, slot: int):
+        self.insert_order(record.order, record.user, slot)
+
+    def handle_order_action_record(self, record: OrderActionRecord, slot: int):
+        if is_one_of_variant(record.action, ['PLACE', 'EXPIRE']):
+            return
+        
+        if is_variant(record.action, 'TRIGGER'):
```

Comment:
Shouldn't this be `Trigger`? tbh we dont use this anymore in ts, might be better to get rid of it in py so it doesn't confuse ppl